scene/gui/code_edit.cpp

```diff
@@ -1329,21 +1329,39 @@ bool CodeEdit::is_drawing_executing_lines_gutter() const {
 
 void CodeEdit::_main_gutter_draw_callback(int p_line, int p_gutter, const Rect2 &p_region) {
 	bool hovering = get_hovered_gutter() == Vector2i(main_gutter, p_line);
-	if (draw_breakpoints && theme_cache.breakpoint_icon.is_valid()) {
+	if (draw_breakpoints && theme_cache.breakpoint_icon.is_valid() && theme_cache.breakpoint_conditional_icon.is_valid() && theme_cache.breakpoint_no_suspend_icon.is_valid() && theme_cache.breakpoint_conditional_no_suspend_icon.is_valid()) {
 		bool breakpointed = is_line_breakpointed(p_line);
 		bool shift_pressed = Input::get_singleton()->is_key_pressed(Key::SHIFT);
 
 		if (breakpointed || (hovering && !is_dragging_cursor() && !shift_pressed)) {
 			int padding = p_region.size.x / 6;
 
 			Color use_color = theme_cache.breakpoint_color;
+			Dictionary dict = get_line_gutter_metadata(p_line, p_gutter);
+			if (!(bool)dict["bp_enabled"]) {
+				use_color = theme_cache.breakpoint_disabled_color;
+			} else if (!((String)dict["bp_print"]).is_empty() && !(bool)dict["bp_suspend"]) {
+				use_color = theme_cache.breakpoint_print_color;
+			}
 			if (hovering && !shift_pressed) {
 				use_color = breakpointed ? use_color.lightened(0.3) : use_color.darkened(0.5);
 			}
 			Rect2 icon_region = p_region;
 			icon_region.position += Point2(padding, padding);
 			icon_region.size -= Point2(padding, padding) * 2;
-			theme_cache.breakpoint_icon->draw_rect(get_canvas_item(), icon_region, false, use_color);
+			if (((String)dict["bp_condition"]).is_empty()) {
```

Comment:
```suggestion
			if (!breakpointed) {
				theme_cache.breakpoint_icon->draw_rect(get_canvas_item(), icon_region, false, use_color);
			} else if (((String)dict["bp_condition"]).is_empty()) {
```
When hovering over a line without a breakpoint, It should use the regular breakpoint icon.

Also because on a regular CodeEdit that isn't part of the ScriptEditor, it shows a broken icon.

![image](https://github.com/user-attachments/assets/c8e6f9fd-a3b1-437e-83a3-237b09345a0c)